examples/medplum-demo-bots/src/dft-transfer-listener/dft-transfer-bot.ts

```diff
@@ -0,0 +1,225 @@
+    insurerOrg = await medplum.createResource({
+      resourceType: 'Organization',
+      identifier: [
+        {
+          system: 'http://terminology.hl7.org/CodeSystem/insurance-plan-identifier',
+          value: insurerId,
+        },
+      ],
+      name: insurerName,
+      type: [
+        {
+          coding: [
+            {
+              system: 'http://terminology.hl7.org/CodeSystem/organization-type',
+              code: 'ins',
+              display: 'Insurance Company',
+            },
+          ],
+        },
+      ],
+    });
+  }
+
+  const coverage = await medplum.createResource<Coverage>({
+    resourceType: 'Coverage',
+    status: 'active',
+    subscriber: createReference(patient),
+    subscriberId: subscriberId,
+    beneficiary: createReference(patient),
+    payor: [createReference(insurerOrg)],
+  });
+
+  return [coverage, insurerOrg];
+}
+
+/**
+ * Creates a Claim resource based on PR1 segments and insurance information
+ * @param medplum - The Medplum client
+ * @param pr1Segments - Array of PR1 segments from an HL7 message
+ * @param patient - The patient to link the claim to
+ * @param insuranceInfo - The coverage and organization to link the claim to
+ * @returns The created Claim resource
+ */
+export async function handleClaimFromPr1s(
+  medplum: MedplumClient,
+  pr1Segments: Hl7Segment[],
+  patient: Patient,
+  insuranceInfo: [Coverage, Organization] | undefined
+): Promise<Claim | undefined> {
+  // Get all procedures
+  const procedures = pr1Segments.map((pr1) => ({
+    code: pr1.getField(3)?.getComponent(1) ?? '', // PR1.3.1 Procedure Code
+    display: pr1.getField(3)?.getComponent(2) ?? '', // PR1.3.2 Procedure Description
+  }));
+
+  if (procedures.length === 0 || !insuranceInfo) {
+    return undefined;
+  }
+
+  const [coverage] = insuranceInfo;
+
+  const claim = await medplum.createResource<Claim>({
+    resourceType: 'Claim',
+    status: 'active',
+    type: {
+      coding: [
+        {
+          system: 'http://terminology.hl7.org/CodeSystem/claim-type',
+          code: 'professional',
+        },
+      ],
+    },
+    use: 'claim',
+    patient: createReference(patient),
+    created: new Date().toISOString(),
+    provider: {
+      display: 'Unknown',
+    },
+    priority: {
+      coding: [
+        {
+          system: 'http://terminology.hl7.org/CodeSystem/processpriority',
+          code: 'normal',
+        },
+      ],
+    },
+    insurance: [
+      {
+        sequence: 1,
+        focal: true,
+        coverage: createReference(coverage),
+      },
+    ],
+    item: procedures.map((proc, index) => ({
+      sequence: index + 1,
+      productOrService: {
+        coding: [
+          {
+            system: 'CPT',
+            code: proc.code,
+            display: proc.display,
+          },
+        ],
+      },
+    })),
+  });
+
+  return claim;
+}
+
+export async function handler(medplum: MedplumClient, event: BotEvent<Hl7Message>): Promise<Hl7Message> {
```

Comment:
Move this "main" function to the top of file. People should be able to read this to get the overview of the Bot, before diving into the details